src/ui.tsx

```diff
@@ -60,6 +60,10 @@ const initialState = {
   progress: (null as number | null),
 };
 
+// Utilize the default state and then overwrite any values that were set and persisted previously
+const persistedPlanOptions = JSON.parse(window.localStorage.getItem("planOptions")) || {};
+initialState.planOptions = {...initialState.planOptions, ...persistedPlanOptions};
```

Comment:
I don't think this works properly for `PaperSize`. In particular, `PaperConfig` calls `state.planOptions.paperSize.isLandscape`, but `JSON.stringify(new PaperSize({x: 100, y: 10}))` returns `{"size":{"x":100,"y":10}}`, so `.isLandscape` will be `null` instead of `true` or `false` (and none of the other methods on `PaperSize` will work either).

I can think of 2 reasonable solutions:

1. After this line, replace `initialState.planOptions.paperSize` by `new PaperSize(initialState.planOptions.paperSize.size)`
```suggestion
initialState.planOptions = {...initialState.planOptions, ...persistedPlanOptions};
initialState.planOptions.paperSize = new PaperSize(initialState.planOptions.paperSize);
```
2. Remove the `PaperSize` class entirely and just use `Vec2`. All of `PaperSize`'s methods can become functions that take a `Vec2` as their first argument.

---

Separately, I'm a little worried about migrations here. If in a future commit we change `planOptions` in a non-backwards-compatible way (e.g. changing the meaning of an option, such as its unit, or changing the way `PaperSize` works) then this has the potential to break the UI in unexpected ways. I think we have a few options here:

1. Don't worry about it; plan options don't change that often, adding new options is the most common change, and this approach works fine for new options, and if it does break then people can just clear their localstorage and carry on.
2. Store a "schema version" number alongside the plan options, which is a hard-coded constant in the code. When we read from local storage, check if the schema version matches; if not then clear localStorage and start from scratch.
3. Same as above, but use the package version (from package.json) as the schema version. This would mean that it would be impossible to forget to update the schema version, with the tradeoff that upgrading saxi would always clear your stored options.


After writing that out, I think I'm actually inclined to go with (1) for now and deal with the problem later if we need to.